### PR TITLE
[profiles] Reduce patch frequency

### PR DESCRIPTION
**Pull request by khewonc**
### What does this PR do?

Nodes were being patched every reconcile when there was no change in labels. This only patches nodes when there is a label change

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: vX.Y.Z
* Cluster Agent: vX.Y.Z

### Describe your test plan

Test in staging:
* Check the workqueue graphs to see if metrics stay consistent
* Check that there aren't an enormous number of patch requests in the k8s requests by method/target object graphs

### Checklist

- [x] PR has at least one valid label: `bug`, `enhancement`, `refactoring`, `documentation`, `tooling`, and/or `dependencies`
- [x] PR has a milestone or the `qa/skip-qa` label

### CODE REVIEW

File: controllers/datadogagent/controller_reconcile_agent_test.go [hunk 1]
```diff
@@ -1534,3 +1536,432 @@ func Test_removeStaleStatus(t *testing.T) {
 		})
 	}
 }
+
+func Test_labelNodesWithProfiles(t *testing.T) {
```
Comment (levan-m) [inline controllers/datadogagent/controller_reconcile_agent_test.go:1540]: Thanks for adding comprehensive test!

**Review by levan-m — APPROVED** [review]

File: controllers/datadogagent/controller_reconcile_agent.go [hunk 2]
```diff
@@ -283,30 +283,45 @@ func (r *Reconciler) labelNodesWithProfiles(ctx context.Context, profilesByNode
...
+		// Remove old profile label key if it is present
+		if _, oldProfileLabelExists := node.Labels[agentprofile.OldProfileLabelKey]; oldProfileLabelExists {
+			labelsToRemove[agentprofile.OldProfileLabelKey] = true
 		}
 
-		// If the profile is not the default one and the label does not exist in
-		// the node, it should be added.
-		if !isDefaultProfile && !profileLabelExists {
-			newLabels[agentprofile.ProfileLabelKey] = profileNamespacedName.Name
+		if len(labelsToRemove) > 0 || len(labelsToAddOrChange) > 0 {
```
Comment (celenechang) [inline controllers/datadogagent/controller_reconcile_agent.go:310]: is there a benefit to having this `||` instead of two separate conditionals?

**Review by celenechang — APPROVED** [review]
one small comment, otherwise looks good!

File: controllers/datadogagent/controller_reconcile_agent.go (hunk 2 above)
Comment (khewonc) [reply controllers/datadogagent/controller_reconcile_agent.go:310]: The first for loop for `labelsToRemove` goes through all the node labels but the second one for `labelsToAddOrChange` doesn't. If we separated them, we'd need to loop through `node.Labels` twice

Comment (celenechang) [reply controllers/datadogagent/controller_reconcile_agent.go:310]: ah i see, thanks!

**Pull request merged**